openreview/venue/process/recruitment_pre_process.js

```diff
@@ -0,0 +1,35 @@
+async function process(client, edit, invitation) {
+
+  const committeeInvitedId = invitation.content.committee_invited_id.value
+  const venueId = invitation.content.venue_id.value
+  const committeeName = invitation.content.committee_id.value.split('/').pop()
+
+  const note = edit.note
+
+  const hash_seed = invitation.content.hash_seed.value
+  const user = decodeURIComponent(note.content.user.value)
+
+  const hashkey = crypto.createHmac('sha256', hash_seed)
+                      .update(user)
+                      .digest('hex');
+
+  if (hashkey != note.content.key.value) {
+    return Promise.reject(new OpenReviewError({ name: 'Error', message: 'Wrong key, please refer back to the recruitment email' }))
+  }
+
+  const { groups } = await client.getGroups({ id: committeeInvitedId, member: user })
```

Comment:
we always check with transitive members because the users could have been invited with different emails or the invited group could have been modified, it is safer to check the membership with transitive members.